### PR TITLE
Added `max_level` param for JSON import feature in the backend

### PR DESCRIPTION
**Pull request by IamEzio**
Fixes part of #2895 

This PR adds `max_level` attribute in the DataFile model and adds it as a param in `/api/db/v0/data_files/` API.
We still have to update the frontend to update `max_level` value while dealing with JSON files (similar to `header` param while dealing with CSV files), but this PR makes the necessary changes in the backend.

**Screenshots**

https://github.com/centerofci/mathesar/assets/89375125/dbee299a-3151-4b53-88de-207e0eb2499f

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by IamEzio** [comment]
@dmos62 PTAL. Thanks!

File: mathesar/imports/json.py [hunk 1]
```diff
@@ -30,22 +30,32 @@ def validate_json_format(data_file_content):
     raise database_api_exceptions.UnsupportedJSONFormat()
 
 
-def get_column_names_from_json(data_file):
+def get_flattened_keys(json_dict, max_level, keys, prefix=''):
+    for key, value in json_dict.items():
+        flattened_key = f"{prefix}{key}"
+        if isinstance(value, dict) and max_level > 0:
+            get_flattened_keys(value, max_level - 1, keys, f"{flattened_key}.")
```
Comment (Anish9901) [inline mathesar/imports/json.py:37]:
```suggestion
def get_flattened_keys(json_dict, max_level, prefix=''):
    keys = []
    for key, value in json_dict.items():
        flattened_key = f"{prefix}{key}"
        if isinstance(value, dict) and max_level > 0:
            keys += get_flattened_keys(value, max_level - 1,  f"{flattened_key}.")
```

**Review by Anish9901 — CHANGES_REQUESTED** [review]
Good work on this Anshuman! There is a tweak that I've suggested below, also I noticed the `max_level` param isn't added to `/api/serializers/data_files.py` without it the request from the frontend won't succeed. Once you have made these changes try to use the DRF client to check if it works correctly.

File: mathesar/imports/json.py (hunk 1 above)
Comment (IamEzio) [reply mathesar/imports/json.py]: Done!

**Comment by IamEzio** [comment]
Thanks @Anish9901, I have updated the PR now. I am not adding the `max_level` param in the serializer yet because I'll have to make changes in the frontend for the `/api/db/v0/data_files/` API requests (since this is not a read-only param and its value needs to be passed from the frontend).
I am hoping to make all the necessary changes in the frontend in a later PR that'll solve #3031.

PTAL at the rest of the suggested changes. Thanks!

**Comment by Anish9901** [comment]
Even if the frontend doesn't make use of the param right now it shouldn't be the concern of the backend, the backend should support it since all the functionality is added. In the serializer, it should be made optional and should default to 0.

**Comment by IamEzio** [comment]
Makes sense. I have updated the PR now. Thanks @Anish9901!

**Review by Anish9901 — APPROVED** [review]
Looks good @IamEzio, Thanks!

**Pull request merged**